### PR TITLE
typescript: handle variables declared within for-loop initializer

### PR DESCRIPTION
**Pull request by evmar**
Rather than specially handling for-loops, instead change around some
other code so our existing processing picks this up.

To understand why this works, consider:
  var x = 3;
  for (var y in ...)
The first is a VariableDeclarationStatement, while both both contain
a VariableDeclaration.

Previously we handled VariableDeclarationStatement specially to extract
defines/binding.  Instead if we handle VariableDeclaration, both forms
are covered.

So this fix for for loops instead only touches VariableDeclaration, and
then to delete a bit of our VariableDeclarationStatement handling.

### CODE REVIEW

**Comment by evmar** [comment]
CC @ayazhafiz

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -1226,6 +1229,9 @@ class Visitor {
         return this.visitExportDeclaration(node as ts.ExportDeclaration);
       case ts.SyntaxKind.VariableStatement:
         return this.visitVariableStatement(node as ts.VariableStatement);
+      case ts.SyntaxKind.VariableDeclaration:
+        this.visitVariableDeclaration(node as ts.VariableDeclaration);
```
Comment (ayazhafiz) [inline kythe/typescript/indexer.ts]:
```suggestion
        return this.visitVariableDeclaration(node as ts.VariableDeclaration);
```

File: kythe/typescript/testdata/for.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export {};
+
+//- @x defines/binding X
+//- X.node/kind variable
+for (const x of [1, 2]) {
+  //- @x ref X
+  x;
+}
```
Comment (ayazhafiz) [inline kythe/typescript/testdata/for.ts]: Can you add some more examples, like standard for loops?

Comment (evmar) [reply kythe/typescript/testdata/for.ts]: Done.  Also it revealed some bugs!  So I fixed those too.

**Comment by evmar** [comment]
@kythe plz merge

**Pull request merged**